sam/bai.go

```diff
@@ -0,0 +1,295 @@
+package sam
```

Comment:
Everything looks good after quick pass, but I am having trouble doing an in-depth review without knowing the specs for the file format.  Can you include a comment at the top of the file to the version/document of the specs you used?

Reply:
I have added a ref to the document.